### PR TITLE
feat(window): implement  on Windows

### PR DESCRIPTION
**Pull request by kleinhammer**
### What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [x] No

### Checklist
- [ ] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [ ] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [x] I have added a convincing reason for adding this feature, if necessary

### Other information

Added implementation of minimizable for Windows

### CODE REVIEW

**Comment by caesar** [comment]
Just tested this on a Windows 11 VM and it isn't working for me – when I run the `minimize` example and hit <kbd>space</kbd> to toggle minimizability, the title bar either disappears (the first time) or (on subsequent toggles) changes to a Windows 7-style title bar but with all the controls non-functional.

I haven't had a chance to investigate why yet (and probably won't get the chance today) but it seems very strange.

**Comment by caesar** [comment]
Sorry, I meant to upload a screen capture of the behaviour I'm observing:

https://user-images.githubusercontent.com/1566516/194124138-64f8a54b-ec15-46c8-af6c-b29d5d4ce9a8.mp4

File: src/platform_impl/windows/window_state.rs [hunk 1]
```diff
@@ -79,6 +79,7 @@ bitflags! {
         const TRANSPARENT      = 1 << 6;
         const CHILD            = 1 << 7;
         const MAXIMIZED        = 1 << 8;
+        const MINIMIZABLE      = 1 << 12;
```
Comment (caesar) [inline src/platform_impl/windows/window_state.rs]: This is a duplicate value (already in use for `MINIMIZED`, see a few lines down).

Per my comments above I suggest suing 18 here, then we can use 19 and 20 for `MAXIMIZABLE` and `CLOSABLE`.

**Review by caesar — DISMISSED** [review]
Hi @kleinhammer, I got a chance to check out what was causing the issue I described above, and seems it's one of the other commits you merged from `dev`, it's not actually caused by your changes.
Could you send a PR with just your own changes, without merging anything else? It makes it easier to see what's going on.

I was able to test your changes by cherry-picking just your commit, but I found that calling `set_minimizable` (pressing <kbd>space</kbd> in the `minimize` example) was actually *minimizing* the window rather than toggling minimizability.

This is because the 12th bit of the `WindowFlags` bitflag, which you've used for the new `MINIMIZABLE` flag, is already in use for the `MINIMIZED` flag.
The highest existing bit that's in use seems to be 17 (after e42ff07190c37b6b8a7808900133838247a730df), so I suggest using 18 for `MINIMIZABLE` and then we can use 19 and 20 for `MAXIMIZABLE` and `CLOSABLE`. 

Does that make sense?

File: src/platform_impl/windows/window_state.rs (hunk 1 above)
Comment (kleincyber) [reply src/platform_impl/windows/window_state.rs]: @caesar Oops. Yeah its a typo. 18 will work! Lets do that.

Comment (kleinhammer) [reply src/platform_impl/windows/window_state.rs]: Edited the bitflags as discussed

Windows 11 Window controls

**Comment by caesar** [comment]
@kleinhammer I've cherry-picked your commits into https://github.com/tauri-apps/tao/pull/574 – it was easier than messing with the merge conflicts from the dev branch. Working well for me now. Cheers!

I'll close this PR. If/when you get a chance to work out maximize / close, could you send another PR with just those changes relative to my branch (without merging dev)? Makes the workflow a bit easier.

**Pull request closed without merging**